packages/shared/src/contexts/FeaturesContext.tsx

```diff
@@ -44,61 +44,64 @@ export type FeaturesContextProviderProps = {
+    onboardingVersion: getFeatureValue(Features.UserOnboardingVersion, flags),
+    onboardingFiltersLayout: getFeatureValue(
+      Features.OnboardingFiltersLayout,
+      flags,
+    ),
+    popularFeedCopy: getFeatureValue(Features.PopularFeedCopy, flags),
+    showCommentPopover: isFeaturedEnabled(Features.ShowCommentPopover, flags),
+    postEngagementNonClickable: isFeaturedEnabled(
+      Features.PostEngagementNonClickable,
+      flags,
+    ),
+    submitArticleOn: isFeaturedEnabled(Features.SubmitArticleOn, flags),
+    canSubmitArticle: isFeaturedEnabled(Features.SubmitArticle, flags),
+    submitArticleSidebarButton: getFeatureValue(
+      Features.SubmitArticleSidebarButton,
+      flags,
+    ),
+    submitArticleModalButton: getFeatureValue(
+      Features.SubmitArticleModalButton,
+      flags,
+    ),
+    postModalByDefault: isFeaturedEnabled(Features.PostModalByDefault, flags),
+    postCardVersion: getFeatureValue(Features.PostCardVersion, flags),
+    postCardShareVersion: getFeatureValue(
+      Features.PostCardShareVersion,
+      flags,
+    ) as ShareVersion,
+    authVersion: getFeatureValue(Features.AuthenticationVersion, flags),
+  };
+};
+
 export const FeaturesContextProvider = ({
   children,
   flags,
 }: FeaturesContextProviderProps): ReactElement => {
-  const steps = getFeatureValue(Features.OnboardingSteps, flags)?.split?.('/');
-  const onboardingSteps = (steps as OnboardingStep[]) || [];
-  const features = useMemo(
-    () => ({
-      flags,
-      onboardingSteps,
-      onboardingMinimumTopics:
-        getFeatureValue(Features.OnboardingMinimumTopics, flags) ?? 0,
-      onboardingVersion: getFeatureValue(Features.UserOnboardingVersion, flags),
-      onboardingFiltersLayout: getFeatureValue(
-        Features.OnboardingFiltersLayout,
-        flags,
-      ),
-      popularFeedCopy: getFeatureValue(Features.PopularFeedCopy, flags),
-      showCommentPopover: isFeaturedEnabled(Features.ShowCommentPopover, flags),
-      postEngagementNonClickable: isFeaturedEnabled(
-        Features.PostEngagementNonClickable,
-        flags,
-      ),
-      submitArticleOn: isFeaturedEnabled(Features.SubmitArticleOn, flags),
-      canSubmitArticle: isFeaturedEnabled(Features.SubmitArticle, flags),
-      submitArticleSidebarButton: getFeatureValue(
-        Features.SubmitArticleSidebarButton,
-        flags,
-      ),
-      submitArticleModalButton: getFeatureValue(
-        Features.SubmitArticleModalButton,
-        flags,
-      ),
-      postModalByDefault: isFeaturedEnabled(Features.PostModalByDefault, flags),
-      postCardVersion: getFeatureValue(Features.PostCardVersion, flags),
-      postCardShareVersion: getFeatureValue(
-        Features.PostCardShareVersion,
-        flags,
-      ) as ShareVersion,
-      authVersion: getFeatureValue(Features.AuthenticationVersion, flags),
-    }),
-    [flags],
-  );
+  const features = useMemo(() => getFeatures(flags), [flags]);
```

Comment:
I think we can do one `useMemo` for both cases. It's not necessary to separate them